src/Core/Queues/IQueue.cs

```diff
@@ -21,10 +21,16 @@ public interface IQueue<T> : IHaveSerializer, IDisposable where T : class {
 
         Task<QueueEntry<T>> DequeueAsync(CancellationToken cancellationToken = default(CancellationToken));
 
+        [Obsolete("Use QueueEntry<T> overload")]

```

Comment:
Aha - should have looked in the base. The only issue with this is to have the poison queue functionality we need the [value of the queue entry](https://github.com/exceptionless/Foundatio/pull/22/files#diff-65336afcf3620424bd815d6ab2436d62R105) to push onto the queue. That would essentially require putting the entire QueueEntry into the cache!